### ISSUE TITLE
Unable to spark_install on Databricks community edition

### ISSUE THREAD

**Issue by tamuanand** [sparklyr/sparklyr#3157]
Hi

I am trying to install spark on Databricks community edition cluster.

I am trying to follow along as listed here - https://therinspark.com/starting.html#starting-prerequisites

What works:
```
1 - signup on Databricks community edition
2 - spinning up a cluster using Databricks 7.3LTS, Spark 3.0.1, Scala 2.12
3 - install.packages("sparklyr") 
4 - library(sparklyr)
```

When I try to do `spark_install("2.3")`, I get this message:
```

Error in download.file(installInfo$packageRemotePath, destfile = installInfo$packageLocalPath, : Error in download.file(installInfo$packageRemotePath, destfile = installInfo$packageLocalPath,  : 
  cannot open URL 'https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz'
Error in download.file(installInfo$packageRemotePath, destfile = installInfo$packageLocalPath,  : 
  cannot open URL 'https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz'
```
How do I get around this?

My sessionInfo:
```
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sparklyr_1.3.1
```

Thanks

**Comment by nviraj** [comment]
Hi, 

The instructions you are referring to are when you are working in local mode and want to install spark in your machine. 
But in your case, spark already comes pre installed when using databricks (3.0.1 edition from your steps)

So all you need to get sparklyr setup is what's mentioned here. 
https://docs.databricks.com/_static/notebooks/sparklyr.html

That being said, there seems to be some issues/ restrictions when trying to download a file in the community edition as seen here: 
https://forums.databricks.com/topics/download+files.html

To check if that is indeed so, you can try the following commands. If this downloads then folder access issue for destination file could be the reason.
```
temp <- tempfile()

download.file("https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz",temp)

file.info(temp, extra_cols = TRUE)
```

TLDR: You don't need that step. Just head to the example notebook.

**Comment by nviraj** [comment]
Ran this on databricks out of curiosity. Reason it's failing is because of 60s timeout in download.file(). It’s happening in non-community edition of databricks as well. 

Attempted commands:
```
temp <- tempfile()
download.file("https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz", temp, method = "auto")
print(file.info(temp, extra_cols = TRUE))
file.remove(temp)
```

Log:
> Error in download.file("https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz",  : 
>   download from 'https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz' failed
> In addition: Warning messages:
> 1: In download.file("https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz",  :
>   downloaded length 48283648 != reported length 226027370
> 2: In download.file("https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz",  :
>   URL 'https://archive.apache.org/dist/spark/spark-2.3.3/spark-2.3.3-bin-hadoop2.7.tgz': Timeout of 60 seconds was reached

Specifically you will need this timeout option before spark_install for the download to finish:
`options(timeout=600)`

yitao-li has investigated this in great detail here:
https://github.com/sparklyr/sparklyr/issues/2409

**Comment by yitao-li** [comment]
Thanks for the great answers @nviraj!

I also modified `sparklyr` sometime ago to temporary set that timeout to 600 while inside `spark_install()` btw precisely for this reason. I recommend switching to `sparklyr` 1.7 which comes with some new features, and also, small improvements such as sensible timeout value for `spark_install()`, etc.

**Comment by tamuanand** [comment]
Thanks @nviraj and @yitao-li
